modules/misc/shell.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, ... }: {
+  options.shell = let enable = config.shell.enableShellIntegration;
+  in {
+    enableBashIntegration = lib.hm.shell.mkBashIntegrationOption enable;
+    enableFishIntegration = lib.hm.shell.mkFishIntegrationOption enable;
+    enableIonIntegration = lib.hm.shell.mkIonIntegrationOption enable;
+    enableNushellIntegration = lib.hm.shell.mkNushellIntegrationOption enable;
+
+    enableShellIntegration = lib.mkOption {
+      default = true;
+
+      description = ''
+        Whether to enable shell integration for all supported shells.
+
+        Individual shell integrations can be overriden with their respective
+        `shell.enable<SHELL>Integration` option. For example, the following
+        declaration globally disables shell integration for Bash:
+
+        ```nix
+        config.shell.enableBashIntegration = false;
+        ```
+      '';
+
+      example = false;
+      type = lib.types.bool;
+    };
+
+    enableZshIntegration = lib.hm.shell.mkZshIntegrationOption enable;
```

Comment:
I think you can put this together with the other `enable<shell>Integration` options above.